### ISSUE TITLE
Change option defaults in plugins

### ISSUE THREAD

**Issue by czosel** [prettier/prettier#3972]
Is it possible to overwrite existing options defaults in plugins?

Example: In the PHP plugin, we'd like to change `tabWidth` from `2` to `4` (https://github.com/prettier/prettier-php/issues/25)

I saw that #3615 might be related, but it seems that it covers only the case where plugins introduce new options.

**Comment by azz** [comment]
I don't think it works with what we have now (happy to be proven wrong though, not 100% sure). Two ways come to mind:

1. Export a new option in the plugins `options` array and fix it in Prettier core to trump the existing option.

2. Add a separate `optionDefaults` that intentionally overrides some default values of an existing **core** option. (`tabWidth`, `useTabs`, or `printWidth`).

I think the latter might be a better way to go.

```js
exports.optionDefaults = {
  tabWith: 2,
};
```

**Comment by ikatyang** [comment]
Option 2 looks good to me but it should only affect the specific plugin, e.g.

   ```js
   // php-plugin.js
   exports.optionDefaults = {
     tabWith: 2,
   };
   ```

   implicitly become something like:

   ```json5
   // .prettierrc
   {
     "overrides": [
       {
         "files": "*.php", // extname from supportLanguages
         "options": {
           "tabWith": 2
         }
       }
     ]
   }
   ```

**Comment by ikatyang** [comment]
For reference, closed #3972 via #3991.
